### PR TITLE
fix: lock body scroll for `Drawer`

### PR DESCRIPTION
**Pull request by mkrl**
[FX-4247](https://toptal-core.atlassian.net/browse/FX-4247)

### Description

Due to the original Picasso implementation, `PicassoRoot` is the default container for all popup dialogs. This causes most popup items to lock scroll on `PicassoRoot` instead of `body`. While it works in most scenarios, when `Drawer` is open on a page with a sticky sidebar and long content that is scrolled down, scroll should be locked on `body`, otherwise it prevents the proper sticky positioning of the sidebar.

### How to test

- [Temploy](https://picasso.toptal.net/fx-4247-wrong-sidebar-appearance-when-drawer-opens)
- Verify the new Happo screenshot has the proper sidebar state
- All the other tests should pass
- Check the client-portal PR https://github.com/toptal/client-portal/pull/7246 (test examples are in the description)

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
|image | image |

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- [n/a] codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4247]: https://toptal-core.atlassian.net/browse/FX-4247?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mkrl** [comment]
@toptal-bot run package:alpha-release

File: cypress/component/Page.spec.tsx [hunk 1]
```diff
@@ -10,6 +17,8 @@ enum TestIds {
   WRAPPER = 'wrapper',
   SIDEBAR_SCROLLABLE_CONTAINER = 'sidebar-scrollable-container',
   MENU_CONTAINER = 'menu-container',
+  DRAWER_BUTTON = 'drawer-button',
+  MENU_ITEM = 'menu_item',
```
Comment (sashuk) [inline cypress/component/Page.spec.tsx:21]: NIT
```suggestion
  MENU_ITEM = 'menu-item',
```

File: cypress/component/Page.spec.tsx [hunk 2]
```diff
@@ -151,6 +202,21 @@ describe('Page', () => {
         variant: 'default/sticky-sidebar-scroll-bottom',
       })
     })
+    it('retains sticky position when Drawer is open', () => {
```
Comment (sashuk) [inline cypress/component/Page.spec.tsx:205]: NIT
```suggestion
  describe('when Drawer is open', () => {
    it('retains sticky position', () => {
```

**Review by sashuk — APPROVED** [review]
Tested the fix in local app, works as intended, code looks good!

https://github.com/toptal/picasso/assets/1390758/7d6763c3-667b-491e-a832-476f8d32ef35

File: cypress/component/Page.spec.tsx (hunk 2 above)
Comment (mkrl) [reply cypress/component/Page.spec.tsx:205]: We already have a describe at the top that kind of makes sense in my mind.
So it goes like this:
- when the sidebar is sticky
  - it sticks to TopBar on scroll
  - it retains sticky position when Drawer is open

Comment (sashuk) [reply cypress/component/Page.spec.tsx:205]: I am usually following the https://toptal-core.atlassian.net/wiki/spaces/ENG/pages/1120830304/Frontend+Testing+-+Unit+testing#Describe in this case

> We have to wrap the use case with `describe` section to accentuate a condition when that test case must happen. 

Anyway, it is a minor comment, just unusual to see a condition in `describe` 🙂

**Pull request merged**